### PR TITLE
no-issue: Improve Admin Panel Password Security

### PR DESCRIPTION
**Pull request by chris-pollard**
### No-Issue:
relates to internal comment on Slack: https://beyondessential.slack.com/archives/C032DCHKTGR/p1665662827255839
### Changes:

- Exclude password from state.
- Disable react Devtools in production

### CODE REVIEW

File: packages/admin-panel/src/index.js [hunk 1]
```diff
@@ -16,6 +16,28 @@ import { TupaiaApi } from './api';
...
+
+      window[globalDevtoolsReactHook][prop] =
+        typeof window[globalDevtoolsReactHook][prop] === 'function' ? () => {} : null;
+    }
+  });
+};
+
+if (process.env.NODE_ENV === 'production') {
+  disableReactDevTools();
+}
```
[the remark this answers is not in the document]
Comment (rohan-bes) [reply packages/admin-panel/src/index.js:39]: No this is probably a good call, thanks @chris-pollard 👍

File: packages/admin-panel/src/authentication/LoginForm.js [hunk 2]
```diff
@@ -108,11 +100,10 @@ const mergeProps = (stateProps, { dispatch }, ownProps) => ({
   ...ownProps,
   ...stateProps,
   onChangeEmailAddress: event => dispatch(changeEmailAddress(event.target.value)),
-  onChangePassword: event => dispatch(changePassword(event.target.value)),
   onChangeRememberMe: event => dispatch(changeRememberMe(event.target.checked)),
-  onLogin: event => {
+  onLogin: (event, password) => {
     event.preventDefault();
-    dispatch(login(stateProps.emailAddress, stateProps.password));
+    dispatch(login(stateProps.emailAddress, password));
```
Comment (rohan-bes) [inline packages/admin-panel/src/authentication/LoginForm.js:106]: Thanks for making this change. Hopefully we can steadily transition most of the app state out of redux and into it's relevant components 👍

File: packages/admin-panel/src/authentication/LoginForm.js [hunk 3]
```diff
@@ -7,15 +7,10 @@ import { TextField, Button, Checkbox } from '@tupaia/ui-components';
...
 import { connect } from 'react-redux';
-import {
-  getEmailAddress,
-  getErrorMessage,
-  getIsLoading,
-  getPassword,
-  getRememberMe,
-} from './selectors';
-import { changeEmailAddress, changePassword, login, changeRememberMe } from './actions';
+import { getEmailAddress, getErrorMessage, getIsLoading, getRememberMe } from './selectors';
```
Comment (rohan-bes) [inline packages/admin-panel/src/authentication/LoginForm.js:12]: Can also delete the `getPassword` selector itself?

File: packages/admin-panel/src/authentication/LoginForm.js [hunk 4]
```diff
@@ -7,15 +7,10 @@ import { TextField, Button, Checkbox } from '@tupaia/ui-components';
...
+import { changeEmailAddress, login, changeRememberMe } from './actions';
```
Comment (rohan-bes) [inline packages/admin-panel/src/authentication/LoginForm.js:13]: Can also delete the `changePassword` action itself?

**Review by rohan-bes — CHANGES_REQUESTED** [review]
Thanks for this @chris-pollard! One request to clean up the associated actions/selectors/reducers for getting/setting the password.

Also wanted to double check, how does this component behave if a user changes the email address while there's something in the password box? I'd expect the password to disappear?

**Comment by chris-pollard** [comment]
> Thanks for this @chris-pollard! One request to clean up the associated actions/selectors/reducers for getting/setting the password.
> 
> Also wanted to double check, how does this component behave if a user changes the email address while there's something in the password box? I'd expect the password to disappear?

@rohan-bes regarding email address change, I have had a look and a change in email address does not seem to affect the state or behaviour of the password input. I had a look at Facebook.com as an example, and it acts the same way. You can change the email address after writing password and the password will not be affected.

**Review by IgorNadj — APPROVED** [review]
Adding tick, already reviewed

**Comment by chris-pollard** [comment]
@rohan-bes for some reason, it seems to still require your approval.

**Comment by chris-pollard** [comment]
@rohan-bes this seems to still require your approval, even though Igor already did.

**Review by rohan-bes — APPROVED** [review]

**Pull request merged**